coordinator/internal/authority/authority.go

```diff
@@ -42,63 +43,59 @@ type metrics struct {
 }
 
 // New creates a new Authority instance.
-func New(hist *history.History, reg *prometheus.Registry, log *slog.Logger) *Authority {
+func New(ctx context.Context, hist *history.History, reg *prometheus.Registry, log *slog.Logger) *Authority {
 	manifestGeneration := promauto.With(reg).NewGauge(prometheus.GaugeOpts{
 		Subsystem: "contrast_coordinator",
 		Name:      "manifest_generation",
 		Help:      "Current manifest generation.",
 	})
 	manifestGeneration.Set(0)
 
-	return &Authority{
+	a := &Authority{
 		hist:   hist,
 		logger: log.WithGroup("mesh-authority"),
 		metrics: metrics{
 			manifestGeneration: manifestGeneration,
 		},
 	}
-}
-
-// syncState ensures that a.state is up-to-date.
-//
-// This function guarantees to include all state updates committed before it was called.
-func (m *Authority) syncState() error {
-	hasLatest, err := m.hist.HasLatest()
-	if err != nil {
-		return fmt.Errorf("probing latest transition: %w", err)
-	}
-	if !hasLatest {
-		// No history yet -> nothing to sync.
-		return nil
-	}
-
-	oldState := m.state.Load()
-	if oldState == nil {
-		// There are transitions in history, but we don't have a local state -> recovery mode.
-		return ErrNeedsRecovery
-	}
-	nextState, err := m.fetchState(oldState.seedEngine)
-	if err != nil {
-		return fmt.Errorf("fetching latest state: %w", err)
-	}
 
-	// Don't update the state if it did not change.
-	if nextState.latest.TransitionHash == oldState.latest.TransitionHash {
-		return nil
-	}
-
-	// Manifest changed, verify that the seedshare owners did not change.
-	if slices.Compare(oldState.manifest.SeedshareOwnerPubKeys, nextState.manifest.SeedshareOwnerPubKeys) != 0 {
-		return fmt.Errorf("can't update from the current manifest to the latest persisted manifest because the seedshare owners changed")
-	}
+	// Set up a watcher that marks the current State stale if it does not correspond to the latest
+	// transition in store.
+
+	go func() {
```

Comment:
time should be fakeable for tests, use kubernetes clock package.

Reply:
Done and test added

Reply:
I find the solution with an variable number of clock arguments which should only be used in tests counter-intuitive.
I'd align this with the usage of the `clock.Ticker` in `NewCachedHTTPSGetter()` in `cached_client.go`, where the caller explicitly passes the wanted clock. 

Another solution might be to just remove the parameter and set the fake clock directly in the struct for the tests.

Reply:
Clock is now an explicit argument to `WatchHistory`.